File: teaching/42-15spring/programming/phi/phi_naive.py
import sys

#The Euclidean algorithm for GCD
def gcd(a,b):
	if b == 0:
		return a
	return gcd(b,a%b)

#Read the number of test cases
T = int(sys.stdin.readline())

#Compute and print the answer to each test case
for t in range(1,T+1):
	n = int(sys.stdin.readline())
	count = 0
	#Try every number 1 to n to see if it is coprime
	for m in range(1,n+1):
		if (gcd(n,m) == 1):
			count += 1
	print "Case #%d: %d" % (t, count)

File: teaching/42-15spring/programming/phi/phi_naive.cpp
#include <iostream>
#include <cstdlib>

using namespace std;
typedef long long ll;

//Euclidean algorithm for GCD
int gcd(int a,int b) {
	if (b == 0) return a;
	else return gcd(b,a%b);
}

//Naive implementation of phi function: try all numbers up to n
//and check if they are coprime using the Euclidean algorithm.
int phi(int n) {
	int count = 0;
	for (int m=1; m<=n; m++) {
		if (gcd(m,n) == 1) count++;
	}
	return count;
}

int main() {
	//Read the number of test cases
	int T; cin >> T;

	//Print the answer to each test case
	for (int t=1; t<=T; t++) {
		int n; cin >> n;
		printf("Case #%d: %d\n",t,phi(n));
	}
}
